sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/RatingDataGenerator.scala

```diff
@@ -46,35 +46,45 @@ object RatingDataGenerator {
       outputPath = args(0)
       numUsers = args(1).toInt
       numProducts = args(2).toInt
-      sparsity = args(3).toDouble
+      numRatings = args(3).toInt
       implicitPrefs = args(4).toBoolean
 
+      require(numRatings <= numUsers * numProducts,
+        s"RatingDataGenerator: numRatings=$numRatings is too large, should be <= numUsers * numProducts"
+      )
+
       println(s"Output Path: $outputPath")
       println(s"Num of Users: $numUsers")
       println(s"Num of Products: $numProducts")
-      println(s"Sparsity: $sparsity")
+      println(s"Num of Ratings: $numRatings")
       println(s"Implicit Prefs: $implicitPrefs")
     } else {
       System.err.println(
-        s"Usage: $RatingDataGenerator <OUTPUT_PATH> <NUM_USERS> <NUM_PRODUCTS> <SPARSITY>  <IMPLICITPREFS>"
+        s"Usage: $RatingDataGenerator <OUTPUT_PATH> <NUM_USERS> <NUM_PRODUCTS> <NUM_RATINGS> <IMPLICITPREFS>"
       )
       System.exit(1)
     }
 
-    val rawData: RDD[Vector] = RandomRDDs.normalVectorRDD(sc, numUsers, numProducts, numPartitions)
-    val rng = new java.util.Random()
-    val data = rawData.map{v =>
-      val a = Array.fill[Double](v.size)(0.0)
-      v.foreachActive{(i,vi) =>
-         if(rng.nextDouble <= sparsity){
-           a(i) = vi
-         }
+    val ratingData = sc.parallelize(1 to numRatings, numPartitions)
+      .mapPartitions { p =>
+        val rng = new java.util.Random()
```

Comment:
Let's use `mapPartitionsWithIndex` and use the partition index as the seed. This way, we can generate same data for different runs and each partition will use different seeds. 
```
import scala.util.hashing
val rng = new java.util.Random((hashing.byteswap32(index))
```